Popup: update Overview demo description (#1996)
DevExpress/devextreme-demos: JSDemos/Demos/Popup/Overview/Angular/description.md

> This demo shows how to show and hide the Popup component, how to fill it with content, specify its position and other configurations. To get started with the DevExtreme Popup component, refer to the following tutorial for step-by-step instructions: [Getting Started with Popup](/Documentation/Guide/UI_Components/Popup/Getting_Started_with_Popup/).

Reviewer: - "shows how to show" - I would try not to repeat similar words close together. 
- "fill it with content" - the word "populate" is a better choice in this context. 
- "specify configurations" - the word "configuration" implies a combination of multiple settings. In this configuration that control behaves like this, and in this configuration it behaves like that... It's the end result of setting individual options. I would use the word "options" or "settings" instead. 

I also don't like the second sentence being in the same paragraph. As I see it, this is a separate thought. The main thought of the first paragraphs is "this is what this demo module shows you". And then you immediately say "you can go to another topic". Should I? Or is it just a note that I can? I assume it is the second option - you may want to go there if you need some step-by-step instructions. But you can also continue with this demo module. This means that I would separate those two thoughts into individual paragraphs. 

```suggestion
This example demonstrates how to show and hide the Popup component, populate it with content, specify its position and other settings. 

To see step-by-step instructions on how to get started with the DevExtreme Popup component, refer to the following tutorial: [Getting Started with Popup](/Documentation/Guide/UI_Components/Popup/Getting_Started_with_Popup/).
```

> This demo shows how to show and hide the Popup component, how to fill it with content, specify its position and other configurations. To get started with the DevExtreme Popup component, refer to the following tutorial for step-by-step instructions: [Getting Started with Popup](/Documentation/Guide/UI_Components/Popup/Getting_Started_with_Popup/).

Reviewer 2: ```suggestion
Learn how to configure the Popup component. This demo details how to show and hide this Popup component, how to fill it with content, how to specify its position and other configurations. To get started with the DevExtreme Popup component, refer to the following step-by-step tutorial: [Getting Started with Popup](/Documentation/Guide/UI_Components/Popup/Getting_Started_with_Popup/).
```
you can also try this format:

Learn how to configure the React Popup component. This demo details how to:

- show and hide this Popup component;
- fill it with content;
- specify its position and other configurations. 

To get started with the DevExtreme React Popup component, refer to the following step-by-step tutorial: [Getting Started with Popup](/Documentation/Guide/UI_Components/Popup/Getting_Started_with_Popup/).

## Show and Hide the Popup

> Use two-way binding between the [visible](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#visible) property and a Boolean variable (`popupVisible` in this demo). The `showInfo(employee)` method sets `popupVisible` to `true` and the Popup appears. You can close the Popup in multiple ways:

Reviewer 2: ```suggestion
Use two-way binding between the [visible](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#visible) property and the Boolean variable (`popupVisible` in this demo). [How does this first sentence relate to the one that follows.] The `showInfo(employee)` method sets `popupVisible` to `true`. This causes the Popup to appear.  To close a Popup, choose one of the following options:
```

Author: The first sentence says that you need to bind the **visible** property to a variable. The second says how to change this variable (by implementing a **showInfo(employee)** method). So I think these two sentences are related.

Reviewer 2: Hi Vlada, :) I know in your mind they are related, but when I am reading - they are not. In fact, your explanation further confuses me, because of how much it differs from the language in the document. Talking about two-way binding is not the same as telling our customer that they need to bind the visible property to a variable. Can you please rewrite this whole section so that those ideas are clear and those sentences logically flow into each other. Thanks.

Author: Okay, I will rewrite it

Author: Done

> Use two-way binding to define the [visible](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#visible) property and bind it to a boolean variable, `popupVisible` in this demo. The `showInfo(employee)` method assigns `true` to the `popupVisible` variable, and the Popup becomes visible. You can close the Popup in multiple ways:

Reviewer: - "use two-way binding ... and bind" - similar words in close succession. 
- "boolean" - should be capitalized, because it comes from a last name. 
- "becomes visible" - "appears" is shorter. Be on the lookout for phrases that combine a common verb plus an adjective. Usually a single verb exist that can be used as a replacement. For example "get bigger" = "grow", "make simpler" = "simplify".

```suggestion
Use two-way binding between the [visible](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#visible) property and a Boolean variable (`popupVisible` in this demo). The `showInfo(employee)` method sets `popupVisible` to `true` and the Popup appears. You can close the Popup in multiple ways:
```

- Built-in close button    

> Enable the [showCloseButton](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#showCloseButton) property to display the close button in the Popup's [title area](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#showTitle).

Reviewer 2: ```suggestion
Enable the [showCloseButton](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#showCloseButton) property to display the Close button in a Popup's [title area](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#showTitle).
```

- Custom close button    

> This demo shows how to add custom buttons to the Popup. One of them uses an [onClick](/Documentation/ApiReference/UI_Components/dxButton/Configuration/#onClick) handler to set the `popupVisible` variable to `false`, and the Popup becomes invisible. To learn more, follow the **Configure the Popup** topic below.

Reviewer: "Follow the ... topic". You can follow a link, or you can follow instructions in a topic.The phrase "follow the ... topic" doesn't seem right to me. 

Plus, it's not a topic... it's a section in this document, right? 

```suggestion
This demo shows how to add custom buttons to the Popup. One of them uses an [onClick](/Documentation/ApiReference/UI_Components/dxButton/Configuration/#onClick) handler to set the `popupVisible` variable to `false`, and the Popup disappears. Refer to the next section (**Configure the Popup**) to learn how you can populate the popup with custom controls.
```

> This demo shows how to add custom buttons to the Popup. One of them uses an [onClick](/Documentation/ApiReference/UI_Components/dxButton/Configuration/#onClick) handler to set the `popupVisible` variable to `false`, and the Popup disappears. Refer to the next section (**Configure the Popup**) to learn how you can populate the popup with custom controls.

Reviewer 2: ```suggestion
This demo shows how to add custom buttons to a Popup. One of these buttons uses an [onClick](/Documentation/ApiReference/UI_Components/dxButton/Configuration/#onClick) handler to set the `popupVisible` variable to `false`, causing the Popup to disappear. Refer to the next section (**Configure the Popup**) to learn how you can populate a popup with custom controls.
```

- On outside click    

> Users can also hide the Popup on outside clicks, if you enable the [hideOnOutsideClick](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#hideOnOutsideClick) property.

Reviewer 2: ```suggestion
Users can also hide a Popup with outside clicks if you enable the [hideOnOutsideClick](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#hideOnOutsideClick) property.
```
what does an outside click mean?

Author: You can either click inside the component or outside of it. So I meant "clicks outside the component", but I think that our customers can understand what an outside click is in the context of a popup window.

Reviewer 2: Don't assume anything. Clearly state all pertinent information.

Author: Okay, I will rewrite this sentence

Author: Done

## Configure the Popup

The Popup inner area is divided into two parts: title area and content area.

- Content area       

> To fill the Popup with content, put a markup inside the component. You can also specify a toolbar at the bottom of the content area. To do this, declare the [toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) in the markup as shown in this demo. Set each item's [toolbar](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#toolbar) property to `bottom`. To learn more about toolbar configuration, refer to the following tutorial: [Getting Started with Toolbar](/Documentation/Guide/UI_Components/Toolbar/Getting_Started_with_Toolbar/)

Reviewer: "a markup" - the word "markup" is an uncountable noun, so should be used without an article if you mean "some markup". 

```suggestion
To populate the Popup with content, add markup inside the component. You can also enable a toolbar at the bottom of the content area. To do this, declare [toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) in the markup as shown in this demo. Set each item's [toolbar](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#toolbar) property to `bottom`. To learn more about toolbar configuration, refer to the following tutorial: [Getting Started with Toolbar](/Documentation/Guide/UI_Components/Toolbar/Getting_Started_with_Toolbar/)
```

> To populate the Popup with content, add markup inside the component. You can also enable a toolbar at the bottom of the content area. To do this, declare [toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) in the markup as shown in this demo. Set each item's [toolbar](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#toolbar) property to `bottom`. To learn more about toolbar configuration, refer to the following tutorial: [Getting Started with Toolbar](/Documentation/Guide/UI_Components/Toolbar/Getting_Started_with_Toolbar/)

Reviewer 2: ```suggestion
To populate a Popup with content, add markup inside the component. [How is this first sentence related to the second? Does enabling a toolbar populate a popup?] You can also enable a toolbar at the bottom of the content area. To do this, declare [toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) in the markup as shown in this demo. Set each item's [toolbar](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#toolbar) property to `bottom`. To learn more about toolbar configuration, refer to the following tutorial: [Getting Started with Toolbar](/Documentation/Guide/UI_Components/Toolbar/Getting_Started_with_Toolbar/)
```

Author: In the UI the toolbar buttons appear inside the Popup window, right below the text/images/whatever content, so I thought that adding buttons also counts as populating the component with content. Like, there's no difference in the UI:
![image](https://user-images.githubusercontent.com/94827090/187605242-f5c42f60-7314-45e0-ac54-b4007603483a.png)

- Title area       

> Set [showTitle](Documentation/ApiReference/UI_Components/dxPopup/Configuration/#showTitle) to `true` and use the [title](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#title) property to specify the caption. You can also add a toolbar in the title area. Set each [toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) item's [toolbar](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#toolbar) property to `top`. If you want to add an item to the overflow menu as shown in this demo, set its [locateInMenu](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#locateInMenu) to `always`.

Reviewer: "toolbarItems item's" - that doesn't read well because of repeated word. 
"set its locateInMenu" - The pronoun (its) is too far from the noun it refers to. Maybe "its" refers to "demo", or "menu"... It's better to be more specific in this case. 

```suggestion
Set [showTitle](Documentation/ApiReference/UI_Components/dxPopup/Configuration/#showTitle) to `true` and use the [title](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#title) property to specify the caption. You can also add a toolbar to the title area. Add [toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) markup and set each item's [toolbar](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#toolbar) property to `top`. If you want to display an item in the overflow menu, as shown in this demo, set the item's [locateInMenu](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#locateInMenu) to `always`.
```

> Set [showTitle](Documentation/ApiReference/UI_Components/dxPopup/Configuration/#showTitle) to `true` and use the [title](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#title) property to specify the caption. You can also add a toolbar to the title area. Add [toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) markup and set each item's [toolbar](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#toolbar) property to `top`. If you want to display an item in the overflow menu, as shown in this demo, set the item's [locateInMenu](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#locateInMenu) to `always`.

Reviewer 2: ```suggestion
Set [showTitle](Documentation/ApiReference/UI_Components/dxPopup/Configuration/#showTitle) to `true` and use the [title](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#title) property to specify a Popup caption. [Same note about relevance here.] You can also add a toolbar to the title area. Add [toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) markup and set each item's [toolbar](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#toolbar) property to `top`. If you want to display an item in the overflow menu, as shown in this demo, set the item's [locateInMenu](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#locateInMenu) to `always`.
```

Author: Okay, this one is a bit trickier. For a user's eye there is no separate toolbar in the title area of a Popup. They kinda merge together. As you can see, both a caption and a button are in the same area at the top of a popup window:
![image](https://user-images.githubusercontent.com/94827090/187606735-ee13f532-53ee-470c-ba0b-79121f93e872.png)
On the left you can see an area occupied by caption. The rest of the space on the right is occupied by toolbar items. And the whole area (that I called title area) is divided from the main Popup content with a line. So i think technically speaking these 2 sentences are related: I tell about how you can populate the left part of the area with a content, and then start to talk about the right part.

Reviewer 2: Hmmm, having a hard time following (due to my own lack of tech understanding). Perhaps, Vova Abadzhev can help rephrase this part with better syntax and flow.

Author: I decided to rewrite these sections completely

## Resize and Position

> To specify the Popup's size, use the [height](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#height) and [width](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#width) properties.

Reviewer 2: ```suggestion
To specify Popup size, use the [height](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#height) and [width](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#width) properties.
```

> In this demo, each Popup's position depends on the image. To position the Popup, specify the **my**, **at**, and **of** properties of the [position](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#position) object. The configuration in the demo reads as follows: "place **my** `center` side **at** the `bottom` side **of** the `#image${employee.ID}` element."

Reviewer: The word "position" repeats three times in the first two sentence. I'm suggesting an alternative without repetition. 

```suggestion
In this demo, each Popup's location is relative to the image. The code sets the **my**, **at**, and **of** properties of the [position](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#position) object. The configuration in the demo reads as follows: "place **my** `center` side **at** the `bottom` side **of** the `#image${employee.ID}` element."
```

> In this demo, each Popup's location is relative to the image. The code sets the **my**, **at**, and **of** properties of the [position](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#position) object. The configuration in the demo reads as follows: "place **my** `center` side **at** the `bottom` side **of** the `#image${employee.ID}` element."

Reviewer 2: When you say the code sets properties - what does it set them to?

Author: I'll change 'set' to 'specify'.

Reviewer 2: Yes, better.

> Use the [container](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#container) property to specify the container to render the Popup in. If you set the [container](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#container) property to a specific element on the page, the [shading](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#shading) applies to this element.

Reviewer 2: ```suggestion
Use the [container](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#container) property to specify in which container to render the Popup. If you set the [container](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#container) property to a specific element on the page, [shading](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#shading) applies to this element.
```

Author: I tried to avoid 'which' here and you added it back 😄 Is my version really that bad?

Reviewer 2: Your version was an incomplete sentence. We can say this instead:
```suggestion
Use the [container](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#container) property to select the container in which you want to render the Popup. If you set the [container](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#container) property to a specific element [WHAT ELEMENT?] on the page, the [shading](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#shading) applies to this element.
```

> Enable the [dragEnabled](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#dragEnabled) property to allow users to move the Popup around the page.

Reviewer: "Enable the dragEnabled" - similar words in close succession. 

```suggestion
Turn on the [dragEnabled](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#dragEnabled) option to allow users to move the Popup around the page.
```

> Turn on the [dragEnabled](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#dragEnabled) option to allow users to move the Popup around the page.

Reviewer 2: ```suggestion
Turn on the [dragEnabled](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#dragEnabled) option to allow users to move a Popup around the page.
```

This example demonstrates how to show and hide the Popup component, populate it with content, specify its position and other settings. 

To see step-by-step instructions on how to get started with the DevExtreme Popup component, refer to the following tutorial: [Getting Started with Popup](/Documentation/Guide/UI_Components/Popup/Getting_Started_with_Popup/).

## Show and Hide the Popup

> Bind the [visible](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#visible) property to a Boolean variable (`popupVisible` in this demo). Then, implement a method that sets this Boolean variable to `true` (the `showInfo(employee)` method in this demo). This causes the Popup to appear. To close a Popup, choose one of the following options:

Reviewer 2: ```suggestion
Bind the [visible](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/#visible) property to a Boolean variable (`popupVisible` in this demo). Next, implement the `showInfo(employee)` method to set this Boolean variable to `true`. This causes the Popup to appear. To close a Popup, choose one of the following options:
```

    
    - Custom   
    Add [toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) markup and set each item's [toolbar](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#toolbar) property to `top`. If you want to display an item in the overflow menu, as shown in this demo, set the item's [locateInMenu](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#locateInMenu) to `always`.

- Content   
To populate the Popup with content, add markup inside the component.

- Bottom toolbar   
To enable bottom toolbar, declare [toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) in the markup as shown in this demo. Set each item's [toolbar](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/#toolbar) property to `bottom`. To learn more about toolbar configuration, refer to the following tutorial: [Getting Started with Toolbar](/Documentation/Guide/UI_Components/Toolbar/Getting_Started_with_Toolbar/)       

> 

Author: ```suggestion
```
